src/core/annotation.js

```diff
@@ -1720,7 +1719,9 @@ class MarkupAnnotation extends Annotation {
   static async createNewAnnotation(xref, annotation, dependencies, params) {
     let oldAnnotation;
     if (annotation.ref) {
-      oldAnnotation = (await xref.fetchIfRefAsync(annotation.ref)).clone();
+      annotation.oldAnnotation = oldAnnotation = (
```

Comment:
Here, and elsewhere throughout this patch/file, passing and accessing `oldAnnotation` both as a separate parameter and as an object-property doesn't feel very readable/maintainable.
Please re-factor all cases to use just *one format* consistently in this code!